### PR TITLE
[MAVEN] #3284: Made modelNamePrefix and -Suffix available through maven

### PR DESCRIPTION
**Pull request by d0x**
As described in #3284

### CODE REVIEW

**Comment by wing328** [comment]
@d0x thanks for the PR.

cc @ePaul

**Comment by ePaul** [comment]
Looks nice. (I didn't test is yet.)

Previously I just passed this argument through the `configOptions` → `additionalProperties` parameter, but this way is certainly nicer.

**Comment by d0x** [comment]
Thanks. 
What exactly you mean by "just passed it through `configOptions` -> `additionalProperties`"?

This wasn't successful for me:

```

 <plugin>
                <groupId>io.swagger</groupId>
                <artifactId>swagger-codegen-maven-plugin</artifactId>
                <version>2.2.0-SNAPSHOT</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <!-- specify the swagger yaml -->
                            <inputSpec>swagger.yaml</inputSpec>

                            <!-- target to generate -->
                            <language>java</language>

                            <!-- pass any necessary config options -->
                            <configOptions>
                                <dateLibrary>java8</dateLibrary>
                                <modelNameSuffix>DTO</modelNameSuffix>
                            </configOptions>

                            <!-- override the default library to jersey2 -->
                            <library>jersey2</library>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

And because it is a parameter in the cli I thought it makes sense to put it there as well.

If there is a way over the `configOptions` which you prefer, I could live with that too.

**Comment by ePaul** [comment]
@d0x I think I used something like this (I don't have a project using this here right now):

                            <configOptions>
                                <additionalProperties>modelNameSuffix=DTO</additionalProperties>
                            </configOptions>

File: modules/swagger-codegen-maven-plugin/src/main/java/io/swagger/codegen/plugin/CodeGenMojo.java [hunk 1]
```diff
@@ -217,6 +225,14 @@ public void execute() throws MojoExecutionException {
             configurator.setLibrary(library);
         }
 
+        if(isNotEmpty(modelNamePrefix)) {
+            configurator.setModelNamePrefix(modelNamePrefix);
+        }
+
+        if(isNotEmpty(modelNameSuffix)) {
+            configurator.setModelNamePrefix(modelNameSuffix);
```
Comment (ePaul) [inline modules/swagger-codegen-maven-plugin/src/main/java/io/swagger/codegen/plugin/CodeGenMojo.java]: You have a copy+paste error here, this should be `configurator.setModelNameSuffix(...)`.

**Comment by ePaul** [comment]
With the fixed configurator call, it does what it pretends to do.

So :+1: depending on fixing what I mentioned.

**Comment by d0x** [comment]
I fixed the c&p error. Sorry for that.

Have a nice sunday.

**Comment by ePaul** [comment]
:+1:  – can be merged from my point of view.

**Comment by wing328** [comment]
@d0x thanks for the contribution. PR merged into master (ready for 2.2.0 stable release)

@ePaul thanks for reviewing the change.

**Pull request merged**